cachecontrol/controller.py

```diff
@@ -323,7 +324,7 @@ def _cache_set(
     def cache_response(
         self,
         request: PreparedRequest,
-        response: HTTPResponse,
+        response_or_ref: HTTPResponse | weakref.ReferenceType[HTTPResponse],
```

Comment:
Is there any chance it would receive a weakref?

Reply:
Yes, the code I changed in adapter.py on line 132 puts it there. This function looks like it is part of public API, so I didn't change it to only accept weakref, only extended it.

I just added a comment below to make clearer why it can be None, 
```
# The weakref can be None only in case the user used streamed request
# and did not consume or close it, and holds no reference to requests.Response.
# In such case, we don't want to cache the response.```